examples/auto-scaling-groups/init.pp

```diff
@@ -0,0 +1,71 @@
+ec2_securitygroup { 'test-sg':
+  ensure      => present,
+  description => 'group for testing autoscaling group',
+  region      => 'sa-east-1',
+}
+
+ec2_launchconfiguration { 'test-lc':
+  ensure          => present,
+  security_groups => ['test-sg'],
+  region          => 'sa-east-1',
+  image_id        => 'ami-67a60d7a',
+  instance_type   => 't1.micro',
+}
+
+ec2_autoscalinggroup { 'test-asg':
+  ensure               => present,
+  min_size             => 2,
+  max_size             => 4,
+  region               => 'sa-east-1',
+  launch_configuration => 'test-lc',
+  availability_zones   => ['sa-east-1b', 'sa-east-1a'],
+}
+
+ec2_scalingpolicy { 'scaleout':
+  ensure             => present,
+  auto_scaling_group => 'test-asg',
+  scaling_adjustment => 30,
+  adjustment_type    => 'PercentChangeInCapacity',
```

Comment:
Normally I would agree here @cmurphy, but @garethr is correct about reduced maint burden.  The AWS service endpoints perform the validation for us.

This is not always the most user-friendly option given it requires an extra network roundtrip, but the size of the API dictates that it's going to be impractically difficult for us to validate "all the things".

If you take a look at the Ruby SDK code, you'll see that they've chosen the same server-side only validation approach (and only have basic schema definitions around the various properties like `AdjustmentType`):
https://github.com/aws/aws-sdk-core-ruby/blob/c51717d4fc68cf7729876ef73b6622b8b51d4c09/aws-sdk-core/apis/AutoScaling.api.json#L786-L791